test/sys/test_socket.rs

```diff
@@ -568,12 +569,142 @@ pub fn test_recv_ipv4pktinfo() {
     thread.join().unwrap();
 }
 
+#[cfg(any(
+    target_os = "freebsd",
+    target_os = "ios",
+    target_os = "macos",
+    target_os = "netbsd",
+    target_os = "openbsd",
+))]
+// qemu doesn't seem to be emulating this correctly in these architectures
+#[cfg_attr(any(
+    target_arch = "mips",
+    target_arch = "mips64",
+    target_arch = "powerpc64",
+), ignore)]
+#[test]
+pub fn test_recvif() {
+    use libc;
+    use nix::ifaddrs::{getifaddrs, InterfaceAddress};
+    use nix::net::if_::*;
+    use nix::sys::socket::sockopt::{Ipv4RecvIf, Ipv4RecvDstAddr};
+    use nix::sys::socket::{bind, AddressFamily, SockFlag, SockType};
+    use nix::sys::socket::{getsockname, setsockopt, socket, SockAddr};
+    use nix::sys::socket::{recvmsg, sendmsg, CmsgSpace, ControlMessage, MsgFlags};
+    use nix::sys::uio::IoVec;
+    use std::io;
+    use std::io::Write;
+    use std::thread;
+
+    fn loopback_v4addr() -> Option<InterfaceAddress> {
```

Comment:
I'm getting deja vu here.  Didn't a recent PR of yours contain something similar or identical to this function?  If so, could you make it common code?